src/Units/UnitModel.java

```diff
@@ -197,6 +197,26 @@ public boolean hasDirectFireWeapon()
     return hasDirect;
   }
 
+  /**
+   * @return True if this UnitModel has at least one weapon that cannot fire after moving.
+   */
+  public boolean hasImmobileWeapon()
+  {
+    boolean hasSiege = false;
+    if(weaponModels != null && weaponModels.size() > 0)
```

Comment:
Technically we don't need to verify `weaponModels.size() > 0` - the for loop can just trivially iterate over zero items.

Reply:
I feel like I ran into problems with it, at some point...